### PR TITLE
Send Spark INFO logs to a file in SparkR tests

### PR DESCRIPTION
**Pull request by shivaram**
This changes also keeps the testthat output on the console (and saves it to R-unit-tests.out)

cc @davies

### CODE REVIEW

File: R/log4j.properties [hunk 1]
```diff
@@ -0,0 +1,28 @@
...
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+# Set everything to be logged to the file target/unit-tests.log
+log4j.rootCategory=INFO, file
```
Comment (davies) [inline R/log4j.properties]: In Spark, we try to do not have a log4j.properties (only templates), when this one will be used? It's good if it's only used in SparkR.

Comment (davies) [inline R/log4j.properties]: If it's only used for tests, should we change the default logging level to WARN?

Comment (shivaram) [inline R/log4j.properties:19]: This file was copied from the core Spark unit tests log4j.properties (https://github.com/apache/spark/blob/master/core/src/test/resources/log4j.properties)

The logging level is still INFO but its logged to a file named R-unit-tests.log. That way the console is not spammed but also you can see the INFO logs for debugging if required. 

We can probably put it inside R/pkg/inst/tests but I wasnt sure if this should be a part of the package.

File: R/run-tests.sh [hunk 2]
```diff
@@ -20,10 +20,10 @@
 FWDIR="$(cd `dirname $0`; pwd)"
 
 FAILED=0
-LOGFILE=unit-tests.log
+LOGFILE=$FWDIR/R-unit-tests.out
 rm -f $LOGFILE
 
-SPARK_TESTING=1 $FWDIR/../bin/sparkR $FWDIR/pkg/tests/run-all.R >$LOGFILE 2>&1
+SPARK_TESTING=1 $FWDIR/../bin/sparkR --driver-java-options "-Dlog4j.configuration=file:$FWDIR/log4j.properties" $FWDIR/pkg/tests/run-all.R 2>&1 | tee $LOGFILE
```
Comment (davies) [inline R/run-tests.sh]: use `tee -a`, or it will be overwritted by `tee`

File: R/run-tests.sh [hunk 3]
```diff
@@ -20,10 +20,10 @@
 FWDIR="$(cd `dirname $0`; pwd)"
 
 FAILED=0
-LOGFILE=unit-tests.log
+LOGFILE=$FWDIR/R-unit-tests.out
```
Comment (davies) [inline R/run-tests.sh]: use `unit-tests.out`? Because it's already in R subdirectory.

File: R/log4j.properties (hunk 1 above)
Comment (davies) [inline R/log4j.properties:19]: I see, thanks, LGTM.

**Comment by davies** [comment]
LGTM, merged.

**Pull request merged**